src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -868,6 +901,19 @@ internal bool TryGetCollectionIterationType(ExpressionSyntax syntax, TypeSymbol
                 var elementType = elementTypeWithAnnotations.Type;
                 Debug.Assert(elementType is { });
 
+                if (collectionTypeKind == CollectionExpressionTypeKind.ImplementsIEnumerable)
+                {
+                    if (!HasCollectionExpressionApplicableConstructor(node.Syntax, targetType, diagnostics))
+                    {
+                        reportedErrors = true;
+                    }
+
+                    if (!HasCollectionExpressionApplicableAddMethod(node.Syntax, targetType, elementType, diagnostics))
+                    {
+                        reportedErrors = true;
```

Comment:
>reportedErrors = true; [](http://example.com/codeflow?start=24&length=22)

It looks like we end up here for ```ListInterfaces_01``` unit-test and report errors that ```HasCollectionExpressionApplicableAddMethod``` reports. I think that the errors look confusing. Consider dropping the diagnostics and reporting a dedicated error instead. Something like "conversion from  collection expression to type '{targetType}' doesn't exist because there is no accessible method that accepts value of type'{elementType}'"

Reply:
Dependencies can be dropped too, since we are reporting an error. Accurate tracking of dependencies is not supported for error scenarios

Reply:
Perhaps we can report "error CS1061: '{targetType}' does not contain a definition for 'Add' and no accessible extension method 'Add' accepting a first argument of type 'string' could be found (are you missing a using directive or an assembly reference?)" when ```HasCollectionExpressionApplicableAddMethod``` doesn't report that itself.

Reply:
It is quite possible that we only need to worry when ```HasCollectionExpressionApplicableAddMethod ``` reports "error CS1950: The best overloaded Add method '{bestCandidate}' for the collection initializer has some invalid arguments"